packages/core/src/middleware/koa-slonik-error-handler.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Slonik Error Types:
+ *
+ *  BackendTerminatedError,
+ *  CheckIntegrityConstraintViolationError,
+ *  ConnectionError,
+ *  DataIntegrityError,
+ *  ForeignKeyIntegrityConstraintViolationError,
+ *  IntegrityConstraintViolationError,
+ *  InvalidConfigurationError,
+ *  InvalidInputError,
+ *  NotFoundError,
+ *  NotNullIntegrityConstraintViolationError,
+ *  StatementCancelledError,
+ *  StatementTimeoutError,
+ *  UnexpectedStateError,
+ *  UniqueIntegrityConstraintViolationError,
+ *  TupleMovedToAnotherPartitionError
+ *
+ * (reference)[https://github.com/gajus/slonik#error-handling]
+ */
+
+import { Middleware } from 'koa';
+import { SlonikError, NotFoundError } from 'slonik';
+
+import RequestError from '@/errors/RequestError';
+import { DeletionError } from '@/errors/SlonikError';
+
+export default function koaSlonikErrorHandler<StateT, ContextT>(): Middleware<StateT, ContextT> {
+  return async (ctx, next) => {
+    try {
+      await next();
+    } catch (error: unknown) {
+      if (!(error instanceof SlonikError)) {
+        throw error;
+      }
+
+      switch (error.constructor) {
+        case DeletionError:
+        case NotFoundError:
+          throw new RequestError({
+            code: 'entity.not_found',
+            status: 404,
```

Comment:
wondering if there is a convenient way to change status code in some situations.

Reply:
Do you mean to change the original staus code of Slonik error types?

Reply:
what if I want to change status (404) to 422

Reply:
hmm so far the error handler itself only catches the Original Slonik error (without any status code), at the very end of the request flow.  In your case, the only thing we have is to catch the error at the route level and return a customized RequestError instead?  Either way, the error needs to be cached at the route level and passed to the handler.